### DISCUSSION TITLE
Obtain forward kinematic equation

### DISCUSSION THREAD

**Discussion by hksBRT** [bulletphysics/bullet3#3656] [General]
If we have a URDF defining our robot, and are able to obtain the right end effector link position by feeding joint values, is there a way to obtain the underlying kinematic equation from pybullet? The relation from the base to the end effector?

**Comment by erwincoumans** [comment]
The forward kinematics are simply a multiplication of matrices. Those matrices are not exposed at the moment.
With "kinematic equation", do you mean exposing those matrices?

**Comment by hksBRT** [comment]
yes

**Comment by itechbear** [comment]
I want this feature too. It would be convenient to do math verifications if pybullet exposes such matrices/vectors.

**Reply by hejia-zhang** [reply]
I want this feature too. Have you found a way to obtain the forward kinematics equation?

**Comment by cmuell89** [comment]
Is there any way FK could be exposed in PyBullet to calculate the pose for an arbitrary link and arbitrary configuration of a robot manipulator?

**Reply by erwincoumans** [reply]
Doesn't resetJointState and getLinkState(ob, link, computeForwardKinematics=True) do this?
https://docs.google.com/document/d/10sXEhzFRSnvFcl3XxNGhnD4N2SedqwdAvK3dsihxVUA/edit#heading=h.2ye70wns7io3

**Reply by cmuell89** [reply]
Hey Erwin. Yes I was literally about to come back and describe this is a viable alternative!
